mllib/src/main/scala/org/apache/spark/ml/fpm/FPGrowth.scala

```diff
@@ -283,17 +283,20 @@ class FPGrowthModel private[ml] (
       .rdd.map(r => (r.getSeq(0), r.getSeq(1)))
       .collect().asInstanceOf[Array[(Seq[Any], Seq[Any])]]
     val brRules = dataset.sparkSession.sparkContext.broadcast(rules)
+    val dt = associationRules.schema("antecedent").dataType
 
-    val dt = dataset.schema($(itemsCol)).dataType
```

Comment:
why change it to `associationRules.schema("antecedent").dataType`?

Reply:
Oh...yeah..we could actually use the original one.